File: Typhoon.Domain/DTOs/Product/ProductDto.cs
using Typhoon.Core.DTOs;
using Typhoon.Domain.DTOs.Category;

namespace Typhoon.Domain.DTOs.Product
{
    public class ProductDto : IDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Brand { get; set; }
        public decimal Price { get; set; }
        public CategoryDto Category { get; set; }
    }
}

File: Typhoon.Client/typhoon-client/src/app/models/productDto.ts
import { CategoryDto } from './categoryDto';

export interface ProductDto {
  id: number;
  name: string;
  description: string;
  brand: string;
  price: number;
  category: CategoryDto;
}

export interface ProductCreateDto {
  name: string;
  description: string;
  brand: string;
  price: number;
  categoryId: number;
}

export interface ProductUpdateDto {
  id: number;
  name: string;
  description: string;
  brand: string;
  price: number;
  categoryId: number;
}
